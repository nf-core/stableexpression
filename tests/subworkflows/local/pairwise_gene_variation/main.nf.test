nextflow_workflow {

    name "Test Workflow PAIRWISE_GENE_VARIATION"
    script "subworkflows/local/pairwise_gene_variation/main.nf"
    workflow "PAIRWISE_GENE_VARIATION"
    tag "subworkflow_pairwise_gene_variation"
    tag "subworkflow"

    test("10 genes") {

        tag "subworkflow_pairwise_gene_variation_10_genes"

        when {
            workflow {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/pairwise_gene_variation/make_chunks/input/counts.head.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("1000 genes") {

        tag "subworkflow_pairwise_gene_variation_1000_genes"

        when {
            workflow {
                """
                ch_counts = Channel.fromPath( '$projectDir/tests/test_data/pairwise_gene_variation/make_chunks/input/counts.parquet', checkIfExists: true)
                input[0] = ch_counts
                """
            }
        }

        then {
            assert workflow.success
            // assert snapshot(workflow.out).match()
        }

    }

}
