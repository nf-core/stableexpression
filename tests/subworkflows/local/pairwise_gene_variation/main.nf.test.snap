{
    "1000 genes": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,ac7ab2bb244b25c15d6fd29bbdef6904"
                ],
                "m_measures": [
                    "m_measures.csv:md5,ac7ab2bb244b25c15d6fd29bbdef6904"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-30T16:17:34.337881721"
    },
    "10 genes": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,597ef653508ddfc81f1c284daca67fbc"
                ],
                "m_measures": [
                    "m_measures.csv:md5,597ef653508ddfc81f1c284daca67fbc"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-30T16:23:31.38356288"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    "m_measures.csv:md5,02faaa46853edf29b168ebb13a9ea361"
                ],
                "m_measures": [
                    "m_measures.csv:md5,02faaa46853edf29b168ebb13a9ea361"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-19T14:51:10.097567803"
    }
}